utils/navigator_gui/navigator_gui/dashboard.py

```diff
@@ -0,0 +1,486 @@
+        self.system_time_frame = self._widget.findChild(QtGui.QFrame, 'system_time_frame')
+        self.system_time_status = self._widget.findChild(QtGui.QLabel, 'system_time_status')
+
+        # Devices table
+        self.device_table = self._widget.findChild(QtGui.QFrame, 'device_table')
+
+        # Control panel buttons
+        toggle_kill_button = self._widget.findChild(QtGui.QPushButton, 'toggle_kill_button')
+        toggle_kill_button.clicked.connect(self.toggle_kill)
+        station_hold_button = self._widget.findChild(QtGui.QPushButton, 'station_hold_button')
+        station_hold_button.clicked.connect(self.station_hold)
+        autonomous_control_button = self._widget.findChild(QtGui.QPushButton, 'autonomous_control_button')
+        autonomous_control_button.clicked.connect(self.select_autonomous_control)
+        rc_control_button = self._widget.findChild(QtGui.QPushButton, 'rc_control_button')
+        rc_control_button.clicked.connect(self.select_rc_control)
+        keyboard_control_button = self._widget.findChild(QtGui.QPushButton, 'keyboard_control_button')
+        keyboard_control_button.clicked.connect(self.select_keyboard_control)
+
+        # Defines the color scheme as QT style sheets
+        self.colors = {
+            "red": "QWidget {background-color:#FF432E;}",
+            "green": "QWidget {background-color:#B1EB00;}",
+            "blue": "QWidget {background-color:#4AA8DB;}",
+            "yellow": "QWidget {background-color:#FDEF14;}"
+        }
+
+    def connect_ros(self):
+        '''
+        Connect ROS nodes, services, and alarms to variables and methods
+        within this class.
+        '''
+        # Attempts to read the battery voltage parameters (sets them to defaults if they have not been set)
+        self.battery_low_voltage = rospy.get_param("/battery_monitor/battery_low_voltage", 22.1)
+        self.battery_critical_voltage = rospy.get_param("/battery_monitor/battery_critical_voltage", 20.6)
+
+        rospy.Subscriber("/battery_monitor", Float32, self.cache_battery_voltage)
+        rospy.Subscriber("/clock", Clock, self.cache_system_time)
+
+        self.wrench_changer = rospy.ServiceProxy('/change_wrench', WrenchSelect)
+        self.station_holder = rospy.ServiceProxy('/lqrrt/station_hold', SetBool)
+
+        self.kb = KillBroadcaster(id='station_hold', description='Resets Pose')
+        self.kill_listener = AlarmListener('kill', self.update_kill_status)
+
+        alarm_broadcaster = AlarmBroadcaster()
+        self.kill_alarm = alarm_broadcaster.add_alarm(
+            name='kill',
+            action_required=True,
+            severity=0
+        )
+
+    def update_kill_status(self, alarm):
+        '''
+        Updates the kill status display when there is an update on the kill
+        alarm. Caches the last displayed kill status to avoid updating the
+        display with the same information twice.
+        '''
+        if (not self.system_time["is_timed_out"]):
```

Comment:
Since a lot of your functions have this not timed out check, you could create a decorator to decorate all the functions with that would check for you. That would be a cool way to do it though this works exactly the same as that would and is easier.